src/main.zig

```diff
@@ -878,11 +878,23 @@ fn getVersionUrl(
     app_data_path: []const u8,
     semantic_version: SemanticVersion,
 ) !DownloadUrl {
-    if (build_options.exe == .zls) return DownloadUrl.initOfficial(std.fmt.allocPrint(
-        arena,
-        "https://builds.zigtools.org/zls-{s}-{}.{s}",
-        .{ os_arch, semantic_version, archive_ext },
-    ) catch |e| oom(e));
+    if (build_options.exe == .zls) {
+        const info_url = try std.fmt.allocPrint(arena, "https://releases.zigtools.org/v1/zls/select-version?compatibility=only-runtime&zig_version={}", .{semantic_version});
```

Comment:
This doesn't work with Zig master versions, since the `+` in the version is unescaped, and this ZLS API seems to expect that:

```
anyzig: error: fetch 'https://releases.zigtools.org/v1/zls/select-version?compatibility=only-runtime&zig_version=0.16.0-dev.1484+d0ba6642b': HTTP response 400 "Bad Request"
```

The working URL in this case is `https://releases.zigtools.org/v1/zls/select-version?compatibility=only-runtime&zig_version=0.16.0-dev.1484%2Bd0ba6642b`.

I applied this fix in [my copy](https://github.com/ianprime0509/anyzig/commit/88747670dbe41da8849b346b6f7149166250cd65) and it works for me:

```suggestion
        const fmt_version = try std.fmt.allocPrint(arena, "{}", .{semantic_version});
        const fmt_version_escaped = try std.mem.replaceOwned(u8, arena, fmt_version, "+", "%2B");
        const info_url = try std.fmt.allocPrint(arena, "https://releases.zigtools.org/v1/zls/select-version?compatibility=only-runtime&zig_version={s}", .{fmt_version_escaped});
```